packages/data-layer/src/data-layer.ts

```diff
@@ -69,8 +99,160 @@ export class DataLayer {
       collections?.googleSheetsUrl === undefined
         ? { type: "hardcoded" }
         : { type: "google-sheet", url: collections.googleSheetsUrl };
+    this.gsIndexerEndpoint = indexer.baseUrl;
+  }
+
+  /**
+   * Allo v1 & v2 builder queries
+   */
+
+  /**
+   * Gets a project by its ID.
+   *
+   * @example
+   * Here is an example:
+   * ```
+   * const project = await dataLayer.getProjectById({
+   *  projectId: "0x1234",
+   *  chainId: 1,
+   *  alloVersion: "allo-v2",
+   * });
+   * ```
+   * @param projectId - the ID of the project to return.
+   * @param chainId - the network ID of the chain.
+   * @param alloVersion - the version of Allo to use.
+   *
+   * @returns v2Project | Project
+   */
+  async getProjectById({
+    projectId,
+    chainId,
+    alloVersion,
+  }: {
+    projectId: string;
+    chainId: number;
+    alloVersion: string;
```

Comment:
```suggestion
    alloVersion: "allo-v1" | "allo-v2";
```

It would be a good idea to make it into a type since i see it's used a lot:

```
type AlloVersion = "allo-v1" |  "allo-v2";
```